### ISSUE TITLE
Allow the C API to be in an arbitrary module.

### ISSUE THREAD

**Issue by Sean1708** [Sean1708/rusty-cheddar#18]

**Comment by Sean1708** [comment]
Also rember to stick a note in the docs reminding people to `pub use` that module.

**Comment by Sean1708** [comment]
Need someway for the module to be specified. Either with a string

```rust
#![plugin(cheddar(module = "path::to::api"))]
```

which would be nice for the user but harder to implement, or with a MetaList

```rust
#![plugin(cheddar(module(path, to, api)))]
```

which is a less nice UX but easier to implement.

**Comment by Sean1708** [comment]
Maybe [`parse_item_from_source_str`](https://manishearth.github.io/rust-internals-docs/syntax/parse/fn.parse_item_from_source_str.html) can help with the first option.

**Comment by Sean1708** [comment]
Closed in a3b3f463dc0fbad2fde4a9fef43e5c0e9482fde9.
